modules/lib/lib-portal/src/main/resources/lib/xp/portal.ts

```diff
@@ -83,15 +83,70 @@ export interface SiteConfig<Config> {
     config: Config;
 }
 
-export interface AssetUrlParams {
+export interface Cookie {
+    value: string;
+    path?: string;
+    domain?: string;
+    maxAge?: number;
+    secure?: boolean;
+    httpOnly?: boolean;
+    sameSite?: 'lax' | 'strict' | 'none' | '';
+}
+
+type LiteralUnion<T extends U, U = string> = T | (U & Record<never, never>);
+
+export type RequestParams = Record<string, string | string[] | undefined>;
+
+export interface Request<Params extends RequestParams = RequestParams,
+    Headers extends Record<string, string | undefined> = Record<string, string | undefined>,
+    Cookies extends Record<string, string | Cookie | undefined> = Record<string, string | Cookie | undefined>,
+    > {
+    method: LiteralUnion<'GET' | 'PUT' | 'POST' | 'DELETE' | 'HEAD' | 'PATCH' | 'OPTIONS' | 'TRACE' | 'CONNECT'>;
```

Comment:
This is actually a list of HTTP verbs you can use in the Java-part of the system:
https://github.com/enonic/xp/blob/52ba85a552c09823bcd499487d824a7a9abe0b98/modules/web/web-api/src/main/java/com/enonic/xp/web/HttpMethod.java#L7